components/gantt/state.md

```diff
@@ -0,0 +1,1158 @@
+                    });
+
+                LastId++;
+            }
+        }
+
+        base.OnInitialized();
+    }
+
+    private void CreateItem(GanttCreateEventArgs args)
+    {
+        var item = args.Item as GanttTask;
+
+        item.Id = LastId++;
+
+        if (args.ParentItem != null)
+        {
+            var parent = (GanttTask)args.ParentItem;
+
+            item.ParentId = parent.Id;
+        }
+
+        Data.Insert(0, item);
+
+        CalculateParentPercentRecursive(item);
+        CalculateParentRangeRecursive(item);
+    }
+
+    private void UpdateItem(GanttUpdateEventArgs args)
+    {
+        var item = args.Item as GanttTask;
+
+        var foundItem = Data.FirstOrDefault(i => i.Id.Equals(item.Id));
+
+        if (foundItem != null)
+        {
+            foundItem.Title = item.Title;
+            foundItem.Start = item.Start;
+            foundItem.End = item.End;
+            foundItem.PercentComplete = item.PercentComplete;
+        }
+    }
+
+    private void DeleteItem(GanttDeleteEventArgs args)
+    {
+        var item = Data.FirstOrDefault(i => i.Id.Equals((args.Item as GanttTask).Id));
+
+        RemoveChildRecursive(item);
+    }
+
+    private void RemoveChildRecursive(GanttTask item)
+    {
+        var children = Data.Where(i => item.Id.Equals(i.ParentId)).ToList();
+
+        foreach (var child in children)
+        {
+            RemoveChildRecursive(child);
+        }
+
+        Data.Remove(item);
+    }
+
+    private void CalculateParentPercentRecursive(GanttTask item)
+    {
+        if (item.ParentId != null)
+        {
+            var parent = GetParent(item);
+
+            var children = GetChildren(parent);
+
+            if (children.Any())
+            {
+                parent.PercentComplete = children.Average(i => i.PercentComplete);
+
+                CalculateParentPercentRecursive(parent);
+            }
+        }
+    }
+
+    private void CalculateParentRangeRecursive(GanttTask item)
+    {
+        if (item.ParentId != null)
+        {
+            var parent = GetParent(item);
+
+            var children = GetChildren(parent);
+
+            if (children.Any())
+            {
+                parent.Start = children.Min(i => i.Start);
+                parent.End = children.Max(i => i.End);
+
+                CalculateParentRangeRecursive(parent);
+            }
+        }
+    }
+
+    private GanttTask GetParent(GanttTask item)
+    {
+        return Data.FirstOrDefault(i => i.Id.Equals(item.ParentId));
+    }
+
+    private IEnumerable<GanttTask> GetChildren(GanttTask item)
+    {
+        return Data.Where(i => item.Id.Equals(i.ParentId));
+    }
+}
+````
+
+
+### Get Current Columns Visibility, Order, Field
+
+The `ColumnStates` property of the `GanttState` object provides you with information about the current state of the Gantt columns. It contains the following properties:
+
+
+Field | Type | Description
```

Comment:
The table is missing vertical borders at both ends. If we agree to use this approach, we should be consistent. (I personally am reluctant about this, as it hinders readability)